samples/snippets/src/main/java/com/example/analytics/QuickstartSample.java

```diff
@@ -0,0 +1,103 @@
+
+ 2. Provide service account credentials using one of the following options:
+ - set the GOOGLE_APPLICATION_CREDENTIALS environment variable, the API
+ client will use the value of this variable to find the service account key
+ JSON file.
+
+ https://cloud.google.com/docs/authentication/production#setting_the_environment_variable
+
+ OR
+
+ - manually pass the path to the service account key JSON file to the API client
+ by specifying the keyFilename parameter in the constructor.
+ https://cloud.google.com/docs/authentication/production#passing_the_path_to_the_service_account_key_in_code
+
+ To run this sample using Maven:
+   cd java-analytics-data/samples/snippets
+   mvn compile
+   mvn exec:java -Dexec.mainClass="QuickstartSample"  -Dexec.args="<GA4 property ID>"
+ */
+
+// [START analytics_data_quickstart]
+
+import com.google.analytics.data.v1alpha.AlphaAnalyticsDataClient;
+import com.google.analytics.data.v1alpha.DateRange;
+import com.google.analytics.data.v1alpha.Dimension;
+import com.google.analytics.data.v1alpha.Entity;
+import com.google.analytics.data.v1alpha.Metric;
+import com.google.analytics.data.v1alpha.Row;
+import com.google.analytics.data.v1alpha.RunReportRequest;
+import com.google.analytics.data.v1alpha.RunReportResponse;
+
+public class QuickstartSample {
+
+  // This is an example snippet that calls the Google Analytics Data API and runs a simple report
+  // on the provided GA4 property id.
+  static void sampleRunReport(String propertyId) throws Exception {
+    // Instantiates a client using default credentials.
+    // See https://cloud.google.com/docs/authentication/production for more information
+    // about managing credentials.
+    try (AlphaAnalyticsDataClient analyticsData = AlphaAnalyticsDataClient.create()) {
+      RunReportRequest request = RunReportRequest.newBuilder()
+          .setEntity(Entity.newBuilder().setPropertyId(propertyId))
+          .addDimensions(
+              Dimension.newBuilder().setName("city"))
+          .addMetrics(Metric.newBuilder().setName("activeUsers"))
+          .addDateRanges(
+              DateRange.newBuilder().setStartDate("2020-03-31").setEndDate("today")).build();
+
+      // Make the request
+      RunReportResponse response = analyticsData.runReport(request);
+
+      System.out.println("Report result:");
+      for (Row row : response.getRowsList()) {
+        System.out.printf("%s, %s%n", row.getDimensionValues(0).getValue(),
+            row.getMetricValues(0).getValue());
+      }
+    }
+  }
+
+  public static void main(String... args) throws Exception {
+    if (args.length != 1) {
+      System.err.println("Arguments: <GA4 property ID>");
+      System.err.println(
+          "A GA4 property id parameter is required to make a query to the"
+              + " Google Analytics Data API.");
+      return;
+    }
+    String propertyId = args[0];
+    sampleRunReport(propertyId);
+  }
+}
```

Comment:
Please follow the [Sample Format][https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/SAMPLE_FORMAT.md] guidelines for writing a sample.

@kurtisvg FYI - care to comment.

Reply:
Les, I believe one guideline clearly violated here is https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/SAMPLE_FORMAT.md#command-line-arguments
I addressed this issue. Did I miss anything else?